### PR TITLE
Feature/profile management

### PR DESCRIPTION
**Pull request by ronin1972**
Add Profile Management of user and organisations. This supports:
1. User registration via Conclave
2. Conclave updates replicating to Jaggaer

### CODE REVIEW

File: cat/CaT-service.yaml [hunk 1]
```diff
@@ -11,6 +11,181 @@ servers:
...
+    patch:
+      tags:
+        - Profile Management
+      description: Updates an existing Tenders API Org with Conclave organisation details (Conclave access only).
+      operationId: app.updateOrg
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: 'https://raw.githubusercontent.com/Crown-Commercial-Service/ccs-conclave-api-definitions/main/OrgProfile/OrgService.yaml#/components/schemas/OrganisationProfileInfo'
```
Comment (tommyb82) [inline cat/CaT-service.yaml:160]: Just to clarify - for the user creation, the endpoint `PUT /tenders/users/{user-id}` has no request body and therefore implies we must look up the user's record in CII - but here we are receiving all the data we need to create the org? Or have I misunderstood..

File: cat/CaT-service.yaml [hunk 2]
```diff
@@ -11,6 +11,181 @@ servers:
...
+        '502':
+          $ref: 'https://raw.githubusercontent.com/Crown-Commercial-Service/ccs-api-definitions-common/main/CCS_OpenAPI_Standards/CCS_Standard_API_Responses.yaml#/components/responses/502BadGateway'
+        '504':
+          $ref: 'https://raw.githubusercontent.com/Crown-Commercial-Service/ccs-api-definitions-common/main/CCS_OpenAPI_Standards/CCS_Standard_API_Responses.yaml#/components/responses/504GatewayTimeout'
+        default:
+          description: successful operation
+      summary: Register user to use Tenders (Jaggaer)
+      
+    get:
+      description: Retrieve the terms and conditions required for CaT and if the user has signed up to them.
```
[the remark this answers is not in the document]
Comment (ronin1972) [reply cat/CaT-service.yaml:74]: Thanks, I will resolve this when I understand the end points we should have. I am rethinking based on other comments.

File: cat/CaT-service.yaml (hunk 1 above)
Comment (ronin1972) [reply cat/CaT-service.yaml:160]: The idea of the PUT is that the service calling it does not have access to the user data. It is held in Conclave User Profile. The users profile will reference and organisation held in the Organisation Profile. The PATCHs are pub/sub on the org profile and user profile data. I need to discuss with you if it is better to have separate end points or use the pub/sub just as a trigger for the tenders API to get the updates itself.

File: cat/CaT-service.yaml [hunk 3]
```diff
@@ -11,6 +11,181 @@ servers:
...
+          $ref: 'https://raw.githubusercontent.com/Crown-Commercial-Service/ccs-api-definitions-common/main/CCS_OpenAPI_Standards/CCS_Standard_API_Responses.yaml#/components/responses/500InternalServerError'
+        '502':
+          $ref: 'https://raw.githubusercontent.com/Crown-Commercial-Service/ccs-api-definitions-common/main/CCS_OpenAPI_Standards/CCS_Standard_API_Responses.yaml#/components/responses/502BadGateway'
+      security:
+        - openID:
+            - read
+            - write
+      summary: Check User registered to use Tenders API (Jaggaer)
+      
+    patch:
```
[the remark this answers is not in the document]
Comment (ronin1972) [reply cat/CaT-service.yaml:111]: The idea of the PUT is that the service calling it does not have access to the user data. It is held in Conclave User Profile. The users profile will reference and organisation held in the Organisation Profile. The PATCHs are pub/sub on the org profile and user profile data. I need to discuss with you if it is better to have separate end points or use the pub/sub just as a trigger for the tenders API to get the updates itself.

File: cat/CaT-service.yaml [hunk 4]
```diff
@@ -11,6 +11,181 @@ servers:
...
+      tags:
+        - Profile Management
+      description: Registers an existing Conclave user to use Tenders API.
+      operationId: app.registerUser
+
+      responses:
+        '201':
+          description: Success (for object creation). Its information is available in the
+            data field at the top level of the response body. The API URL where
+            the object can be retrieved is also returned in the Location header of the response.
```
[the remark this answers is not in the document]
Comment (ronin1972) [reply cat/CaT-service.yaml:29]: Not sure I understand this point. I will try to chat with you an make sure I understand.

[the remark this answers is not in the document]
Comment (ronin1972) [reply cat/CaT-service.yaml:29]: I think it comes from this:
https://stackoverflow.com/questions/1860645/create-request-with-post-which-response-codes-200-or-201-and-content

File: cat/CaT-service.yaml (hunk 1 above)
Comment (ronin1972) [reply cat/CaT-service.yaml:160]: @tommyb82 I ave also thought about this a bit more. The PATCH is likely to be authenticated as a system i.e. the Adapter Service where as the PUT is going to be authenticated as a user i.e. the user should be the owner of the user-id or an admin user. Currently the user-id must always be that of the requesting user.

File: cat/CaT-service.yaml [hunk 5]
```diff
@@ -19,10 +19,34 @@ paths:
...
 
       responses:
+        '200':
+          description: Success (for object update).
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  userAction:
```
Comment (tommyb82) [inline cat/CaT-service.yaml:33]: Is `userAction` redundant now that a `200` response represents a successful update and a `201` a successful create?

**Review by tommyb82 — APPROVED** [review]
Looks good - just that one final comment / consideration 👍

**Pull request merged**